### PR TITLE
[SHELL32] Initial implement of Folder Customize

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
This PR is an initial implementation of Folder Customize.
JIRA issue: [CORE-11407](https://jira.reactos.org/browse/CORE-11407)

## Proposed changes

- Add IDD_FOLDER_CUSTOMIZE resource dialog.
- Add a "Customize" page to Folder property sheet.

### CODE REVIEW

**Comment by katahiromz** [comment]
AFTER:
folder-customize

**Comment by katahiromz** [comment]
Implemented changing folder icons:
folder-after

File: dll/win32/shell32/dialogs/filedefext.cpp [hunk 1]
```diff
@@ -779,6 +779,21 @@ CFileDefExt::InitVersionPage(HWND hwndDlg)
     return TRUE;
 }
 
+BOOL
+CFileDefExt::InitFolderCustomizePage(HWND hwndDlg)
+{
+    /* Attach file version to dialog window */
+    SetWindowLongPtr(hwndDlg, DWLP_USER, (LONG_PTR)this);
```
[the remark this answers is not in the document]
Comment (katahiromz) [reply dll/win32/shell32/dialogs/filedefext.cpp]: It's difficult for me.

File: dll/win32/shell32/dialogs/filedefext.cpp [hunk 2]
```diff
@@ -885,11 +885,256 @@ CFileDefExt::VersionPageProc(HWND hwndDlg, UINT uMsg, WPARAM wParam, LPARAM lPar
...
+
+        // change desktop.ini attributes
+        attrs = GetFileAttributesW(szIniFile);
+        attrs |= FILE_ATTRIBUTE_SYSTEM | FILE_ATTRIBUTE_HIDDEN;
+        SetFileAttributesW(szIniFile, attrs);
+
+        // change folder attributes
+        attrs = GetFileAttributesW(m_wszPath);
+        attrs |= FILE_ATTRIBUTE_SYSTEM;
+        SetFileAttributesW(m_wszPath, attrs);
```
Comment (wjk) [inline dll/win32/shell32/dialogs/filedefext.cpp]: Why are you setting the folder to +system here? Is this what Windows does? It just seems like an odd change to make.

Comment (katahiromz) [reply dll/win32/shell32/dialogs/filedefext.cpp]: 2K3:
attrib
The folder is +readonly. The desktop.ini is +system +hidden.

Comment (katahiromz) [reply dll/win32/shell32/dialogs/filedefext.cpp]: OK.

Comment (HBelusca) [reply dll/win32/shell32/dialogs/filedefext.cpp]: Does this also apply for the other folders like, "My Documents" or "My Images" ?

Comment (katahiromz) [reply dll/win32/shell32/dialogs/filedefext.cpp]: Some special folders have the system attribute. The system attribute or read-only attribute are required for custom folder icon.

**Pull request merged**